### ISSUE TITLE
sort-imports should enforce consistent whitespace between imports

### ISSUE THREAD

**Issue by lstrzebinczyk** [eslint/eslint#9991]
**What rule do you want to change?**

sort-imports

**Does this change cause the rule to produce more or fewer warnings?**

This change would cause the rule to produce more warnings.

**How will the change be implemented? (New option, new default behavior, etc.)?**

Ideally this should be default behavior, but this would be backwards incompatible. Doable way would be to introduce new option set by default to false. Only when set to true (or some string), rule would enforce consistent whitespace.

**Please provide some example code that this change will affect:**

```js
// option consistentWhitespace: true
// this code is incorrect:

import "aaa";
import "bbb";
import "./aaa";
import "./bbb";
import _ from "xyz";
import a from "a";
import A from "A";
import B, {b} from "B";
import {C} from C;
import __ from "./xyz";
import aa from "./a";
import AA from "./A";
import BB, {bb} from "./B"
import {CC} from "./C"

// this code is correct:

import "aaa";
import "bbb";

import "./aaa";
import "./bbb";

import _ from "xyz";
import a from "a";
import A from "A";
import B, {b} from "B";
import {C} from C;

import __ from "./xyz";
import aa from "./a";
import AA from "./A";
import BB, {bb} from "./B"
import {CC} from "./C"

```

**What does the rule currently do for this code?**

The rule currently is not concerned with whitespace between imports.

**What will the rule do after it's changed?**

After the rule is changed it will expose additional option, which, when set, will enforce consistent whitespace between imports. Rules concerning where whitespace between imports should be set seem to be established in text editor plugins. The example I provided come from https://marketplace.visualstudio.com/items?itemName=amatiasq.sort-imports

**Comment by not-an-aardvark** [comment]
Thanks for the proposal. I'm not sure it would make sense for `sort-imports` to also check whitespace, because the name implies that it should just care about sorting. Does the [`padding-line-between-statements`](https://eslint.org/docs/rules/padding-line-between-statements) work for this use case?

**Comment by KillaPL** [comment]
The proposal is about whitespace between various types of import statements, `padding-line-between-statements` only recognizes imports as one group. I don't think it would work for this case.

If not as part of `sort-imports`, does it make sense in different context? We can implement this as custom rule for ourselves in my company, but I thought this is general enough and de-facto-standard due to plugins enforcing it, and there would be value in implementing rule within eslint itself.

**Comment by KillaPL** [comment]
@not-an-aardvark How does this work, are we discussing about approval of the idea first? Should I provide MR with the code implementing this? I want to move with this 100% according to your rulebook : )

**Comment by kaicataldo** [comment]
I apologize that we lost track of this.

I agree that `sort-imports` should not be in the business of checking whitespace. If you'd still like to push for this, it would be really helpful to have a concrete proposal. It's hard for us to evaluate what to do here without one.

**Comment by lstrzebinczyk** [comment]
It was a long shot anyway, so I will not be pushing for this: ). I'm closing the issue, thanks.
